uniffi_meta/src/child_types.rs

```diff
@@ -0,0 +1,107 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use crate::Type;
+
+/// Iterate over all child types as mutable references
+///
+/// This is used to find and convert external type
+pub trait ChildTypesMut {
+    fn child_types_mut(&mut self) -> Vec<&mut Type>;
+}
+
+impl ChildTypesMut for crate::Metadata {
```

Comment:
I've nearly turned [`TypeIterator`](https://github.com/mozilla/uniffi-rs/blob/main/uniffi_bindgen/src/interface/types/mod.rs#L127) into a trait before, which would make even more sense here I think? Is there any reason to not do that here? ie, define the trait as a more general `IterTypes`and later we might let it be used in more places (eg, in that example above)

Although really, I don't quite understand how this differs from the above example? I guess because we don't actually have a struct to represent `Type::External`?

(hrmph - the fact this is `mut` also seems odd. I'll look at this in more detail again soon)